var documenterSearchIndex = {"docs":
[{"location":"functions/#FBCModelTests.jl-functions","page":"Reference","title":"FBCModelTests.jl functions","text":"","category":"section"},{"location":"functions/","page":"Reference","title":"Reference","text":"Modules = [FBCModelTests]\nPages = [\"FBCModelTests.jl\"]","category":"page"},{"location":"functions/#FBCModelTests.FBCModelTests","page":"Reference","title":"FBCModelTests.FBCModelTests","text":"module FBCModelTests\n\nA collection of tests for constraint-based flux metabolic models.\n\n\n\n\n\n","category":"module"},{"location":"functions/#FROG","page":"Reference","title":"FROG","text":"","category":"section"},{"location":"functions/","page":"Reference","title":"Reference","text":"Modules = [FBCModelTests, FBCModelTests.FROG]\nPages = [\"frog.jl\", \"frog/structs.jl\", \"frog/frontend.jl\"]","category":"page"},{"location":"functions/#FBCModelTests.FROG","page":"Reference","title":"FBCModelTests.FROG","text":"module FROG\n\nA collection of reproducibility checks for constraint-based metabolic models together with a report generator and tester.\n\nSee https://www.ebi.ac.uk/biomodels/curation/fbc for details.\n\n\n\n\n\n","category":"module"},{"location":"functions/#FBCModelTests.FROG.ObjectiveValue","page":"Reference","title":"FBCModelTests.FROG.ObjectiveValue","text":"\n\n\n\n","category":"type"},{"location":"functions/#FBCModelTests.FROG.FROGMetadata","page":"Reference","title":"FBCModelTests.FROG.FROGMetadata","text":"mutable struct Dict{String, String} <: AbstractDict{String, String}\n\n\n\n\n\n","category":"type"},{"location":"functions/#FBCModelTests.FROG.FROGObjectiveReport","page":"Reference","title":"FBCModelTests.FROG.FROGObjectiveReport","text":"struct FROGObjectiveReport\n\nFields\n\noptimum::Union{Nothing, Float64}\nreactions::Dict{String, FBCModelTests.FROG.FROGReactionReport}\ngene_deletions::Dict{String, Union{Nothing, Float64}}\n\n\n\n\n\n","category":"type"},{"location":"functions/#FBCModelTests.FROG.FROGReactionReport","page":"Reference","title":"FBCModelTests.FROG.FROGReactionReport","text":"struct FROGReactionReport\n\nFields\n\nflux::Union{Nothing, Float64}\nvariability_min::Union{Nothing, Float64}\nvariability_max::Union{Nothing, Float64}\ndeletion::Union{Nothing, Float64}\n\n\n\n\n\n","category":"type"},{"location":"functions/#FBCModelTests.FROG.FROGReportData","page":"Reference","title":"FBCModelTests.FROG.FROGReportData","text":"mutable struct Dict{String, FBCModelTests.FROG.FROGObjectiveReport} <: AbstractDict{String, FBCModelTests.FROG.FROGObjectiveReport}\n\n\n\n\n\n","category":"type"},{"location":"functions/#FBCModelTests.FROG.compare_reports-Tuple{String, String}","page":"Reference","title":"FBCModelTests.FROG.compare_reports","text":"compare_reports(\n    report_dir_a::String,\n    report_dir_b::String\n) -> Union{Test.FallbackTestSet, Test.DefaultTestSet}\n\n\nA simple wrapper for comparing 2 previously generated reports in their respective directories.\n\n\n\n\n\n","category":"method"},{"location":"functions/#FBCModelTests.FROG.generate_report-Tuple{String}","page":"Reference","title":"FBCModelTests.FROG.generate_report","text":"generate_report(\n    filename::String;\n    report_dir,\n    optimizer,\n    workers,\n    basefilename\n)\n\n\nA complete function for one-shot generation of FROG reports. Use ReportGenerators.generate_report_data, ReportGenerators.generate_metadata and ReportIO.save_report for finer control of the process.\n\n\n\n\n\n","category":"method"},{"location":"functions/#Reading-and-writing-reports","page":"Reference","title":"Reading and writing reports","text":"","category":"section"},{"location":"functions/","page":"Reference","title":"Reference","text":"Modules = [FBCModelTests.FROG.ReportIO]\nPages = [\"frog/io.jl\"]","category":"page"},{"location":"functions/#FBCModelTests.FROG.ReportIO","page":"Reference","title":"FBCModelTests.FROG.ReportIO","text":"module ReportIO\n\nFunctions for reading and writing FROG reports.\n\n\n\n\n\n","category":"module"},{"location":"functions/#FBCModelTests.FROG.ReportIO.load_report-Tuple{String}","page":"Reference","title":"FBCModelTests.FROG.ReportIO.load_report","text":"load_report(\n    report_dir::String\n) -> NamedTuple{(:metadata, :report), Tuple{Dict{String, String}, Dict{String, FBCModelTests.FROG.FROGObjectiveReport}}}\n\n\nReverse of save_report.\n\n\n\n\n\n","category":"method"},{"location":"functions/#FBCModelTests.FROG.ReportIO.save_report-Tuple{Dict{String, FBCModelTests.FROG.FROGObjectiveReport}, Dict{String, String}}","page":"Reference","title":"FBCModelTests.FROG.ReportIO.save_report","text":"save_report(\n    r::Dict{String, FBCModelTests.FROG.FROGObjectiveReport},\n    metadata::Dict{String, String};\n    report_dir,\n    basefilename\n)\n\n\nWrite the contents of FROGReportData to the 4 TSV files as specified by FROG standard, and additionally write the metadata into the JSON file.\n\n\n\n\n\n","category":"method"},{"location":"functions/#Generating-and-testing-the-reports","page":"Reference","title":"Generating and testing the reports","text":"","category":"section"},{"location":"functions/","page":"Reference","title":"Reference","text":"Modules = [FBCModelTests.FROG.ReportGenerators, FBCModelTests.FROG.ReportTests]\nPages = [\"frog/report.jl\"]","category":"page"},{"location":"functions/#FBCModelTests.FROG.ReportGenerators","page":"Reference","title":"FBCModelTests.FROG.ReportGenerators","text":"module ReportGenerators\n\nFunctions for generating FROG report data and metadata file contents.\n\n\n\n\n\n","category":"module"},{"location":"functions/#FBCModelTests.FROG.ReportGenerators.ResetObjective","page":"Reference","title":"FBCModelTests.FROG.ReportGenerators.ResetObjective","text":"struct ResetObjective <: COBREXA.ModelWrapper\n\nFields\n\nmodel::COBREXA.MetabolicModel\nobjective::SparseArrays.SparseVector{Float64, Int64}\n\n\n\n\n\n","category":"type"},{"location":"functions/#FBCModelTests.FROG.ReportGenerators.frog_objective_report-Tuple{COBREXA.SBMLModel, String}","page":"Reference","title":"FBCModelTests.FROG.ReportGenerators.frog_objective_report","text":"frog_objective_report(\n    sbml_model::COBREXA.SBMLModel,\n    objective::String;\n    optimizer,\n    workers\n)\n\n\nGenerate a FROGObjectiveReport containing the reproducibility data for a single objective in the SBML model.\n\n\n\n\n\n","category":"method"},{"location":"functions/#FBCModelTests.FROG.ReportGenerators.generate_metadata-Tuple{String}","page":"Reference","title":"FBCModelTests.FROG.ReportGenerators.generate_metadata","text":"generate_metadata(filename::String; optimizer, basefilename)\n\n\n\n\n\n\n","category":"method"},{"location":"functions/#FBCModelTests.FROG.ReportGenerators.generate_report_data-Tuple{COBREXA.SBMLModel}","page":"Reference","title":"FBCModelTests.FROG.ReportGenerators.generate_report_data","text":"generate_report_data(\n    model::COBREXA.SBMLModel;\n    optimizer,\n    workers\n)\n\n\nGenerate FROGReportData for a model.\n\n\n\n\n\n","category":"method"},{"location":"functions/#FBCModelTests.FROG.ReportTests","page":"Reference","title":"FBCModelTests.FROG.ReportTests","text":"module ReportTests\n\nFunction for testing the compatibility of FROG report data.\n\n\n\n\n\n","category":"module"},{"location":"functions/#FBCModelTests.FROG.ReportTests.test_metadata_compatibility-Tuple{Dict{String, String}, Dict{String, String}}","page":"Reference","title":"FBCModelTests.FROG.ReportTests.test_metadata_compatibility","text":"test_metadata_compatibility(\n    a::Dict{String, String},\n    b::Dict{String, String}\n)\n\n\n\n\n\n\n","category":"method"},{"location":"functions/#FBCModelTests.FROG.ReportTests.test_report_compatibility-Tuple{Dict{String, FBCModelTests.FROG.FROGObjectiveReport}, Dict{String, FBCModelTests.FROG.FROGObjectiveReport}}","page":"Reference","title":"FBCModelTests.FROG.ReportTests.test_report_compatibility","text":"test_report_compatibility(\n    a::Dict{String, FBCModelTests.FROG.FROGObjectiveReport},\n    b::Dict{String, FBCModelTests.FROG.FROGObjectiveReport};\n    absolute_tolerance,\n    relative_tolerance\n) -> Union{Test.FallbackTestSet, Test.DefaultTestSet}\n\n\n\n\n\n\n","category":"method"},{"location":"functions/#MEMOTE","page":"Reference","title":"MEMOTE","text":"","category":"section"},{"location":"functions/","page":"Reference","title":"Reference","text":"Modules = [FBCModelTests.Memote]\nPages = [\"memote.jl\", \"memote/frontend.jl\"]","category":"page"},{"location":"functions/#FBCModelTests.Memote","page":"Reference","title":"FBCModelTests.Memote","text":"module Memote\n\nThis package contains a collection of tests based on Memote. See Lieven, C., Beber, M.E., Olivier, B.G. et al. MEMOTE for standardized genome-scale metabolic model testing. Nat Biotechnol 38, 272–276 (2020). https://doi.org/10.1038/s41587-020-0446-y for details.\n\n\n\n\n\n","category":"module"},{"location":"functions/#FBCModelTests.Memote.generate_report-Tuple{COBREXA.MetabolicModel, Any}","page":"Reference","title":"FBCModelTests.Memote.generate_report","text":"generate_report(\n    model::COBREXA.MetabolicModel,\n    optimizer;\n    config\n) -> Dict\n\n\nGenerate a machine-readable report of model properties that can be used as a baseline for checking the quality of the model. Suitable for being exported as JSON.\n\n\n\n\n\n","category":"method"},{"location":"functions/#FBCModelTests.Memote.generate_report-Tuple{String, Any}","page":"Reference","title":"FBCModelTests.Memote.generate_report","text":"generate_report(\n    filename::String,\n    optimizer;\n    kwargs...\n) -> Dict\n\n\nOverload of generate_report that works directly on a given filename.\n\n\n\n\n\n","category":"method"},{"location":"functions/#FBCModelTests.Memote.run_tests-Tuple{COBREXA.MetabolicModel, Any}","page":"Reference","title":"FBCModelTests.Memote.run_tests","text":"run_tests(\n    model::COBREXA.MetabolicModel,\n    optimizer;\n    config,\n    filename\n) -> Union{Test.FallbackTestSet, Test.DefaultTestSet}\n\n\nRun a MEMOTE-like test suite on model using optimizer to solve optimization problems; some basic parameters and thresholds are taken from config.\n\nSome of the tests internally convert the input model to StandardModel of COBREXA; supplying a StandardModel may thus increase efficiency of the whole process.\n\n\n\n\n\n","category":"method"},{"location":"functions/#FBCModelTests.Memote.run_tests-Tuple{String, Any}","page":"Reference","title":"FBCModelTests.Memote.run_tests","text":"run_tests(\n    filename::String,\n    optimizer;\n    kwargs...\n) -> Union{Test.FallbackTestSet, Test.DefaultTestSet}\n\n\nOverload of run_tests that works directly with a file.\n\n\n\n\n\n","category":"method"},{"location":"functions/#MEMOTE-utilities","page":"Reference","title":"MEMOTE utilities","text":"","category":"section"},{"location":"functions/","page":"Reference","title":"Reference","text":"Modules = [FBCModelTests.Memote.Utils, FBCModelTests.Memote.Config]\nPages = [\"memote/utils.jl\", \"memote/config.jl\"]","category":"page"},{"location":"functions/#FBCModelTests.Memote.Utils","page":"Reference","title":"FBCModelTests.Memote.Utils","text":"module  Utils\n\nMiscellaneous functions used by memote style tests, not typically user facing.\n\n\n\n\n\n","category":"module"},{"location":"functions/#FBCModelTests.Memote.Utils._has_sensible_gpr-Tuple{COBREXA.MetabolicModel, Any}","page":"Reference","title":"FBCModelTests.Memote.Utils._has_sensible_gpr","text":"_has_sensible_gpr(model::COBREXA.MetabolicModel, rid) -> Any\n\n\nInternal helper function that determines if a reaction has a gene reaction rule and that each gene in the rule is contained in the model.\n\n\n\n\n\n","category":"method"},{"location":"functions/#FBCModelTests.Memote.Utils._probably_transport_reaction-Tuple{COBREXA.MetabolicModel, Any, Any}","page":"Reference","title":"FBCModelTests.Memote.Utils._probably_transport_reaction","text":"_probably_transport_reaction(\n    model::COBREXA.MetabolicModel,\n    rid,\n    test_annotation\n) -> Bool\n\n\nDetermine if a reaction is probably a transport reaction by checking if:\n\nit has sbo annotations corresponding to a transport reaction\nthe reaction contains metabolites from at least 2 different compartments\nif at least 1 metabolite does not undergo a chemical transformation (via formula or annotation checks)\n\nNote, PTS type transport reactions will be missed if they do not have sbo annotations. This test may yield false negatives.\n\n\n\n\n\n","category":"method"},{"location":"functions/#FBCModelTests.Memote.Utils.to_element-Tuple{String}","page":"Reference","title":"FBCModelTests.Memote.Utils.to_element","text":"to_element(x::String) -> PeriodicTable.Element\n\n\nReturn the chemical element of x.\n\n\n\n\n\n","category":"method"},{"location":"functions/#FBCModelTests.Memote.Config","page":"Reference","title":"FBCModelTests.Memote.Config","text":"module Config\n\nModule housing the configuration parameters for the memote-style tests.\n\n\n\n\n\n","category":"module"},{"location":"functions/#FBCModelTests.Memote.Config.AnnotationConfig","page":"Reference","title":"FBCModelTests.Memote.Config.AnnotationConfig","text":"mutable struct AnnotationConfig\n\nFields\n\ngene_annotation_keywords::Vector{String}\ngene_annotation_regexes::Dict{String, Regex}\nmetabolite_annotation_keywords::Vector{String}\nmetabolite_annotation_regexes::Dict{String, Regex}\nreaction_annotation_keywords::Vector{String}\nreaction_annotation_regexes::Dict{String, Regex}\nminimum_fraction_database_conformity::Float64\nminimum_fraction_database_annotations::Float64\n\n\n\n\n\n","category":"type"},{"location":"functions/#FBCModelTests.Memote.Config.BasicConfig","page":"Reference","title":"FBCModelTests.Memote.Config.BasicConfig","text":"mutable struct BasicConfig\n\nFields\n\nminimum_metabolic_coverage::Float64\nminimum_growth_rate::Float64\nmaximum_growth_rate::Float64\noptimizer_modifications::Vector{Function}\n\n\n\n\n\n","category":"type"},{"location":"functions/#FBCModelTests.Memote.Config.BiomassConfig","page":"Reference","title":"FBCModelTests.Memote.Config.BiomassConfig","text":"mutable struct BiomassConfig\n\nFields\n\nbiomass_strings::Vector{String}\ngrowth_metabolites::Dict{String, String}\nignored_precursors::Vector{String}\nessential_precursors::Dict{String, String}\nminimum_growth_rate::Float64\noptimizer_modifications::Vector{Function}\n\n\n\n\n\n","category":"type"},{"location":"functions/#FBCModelTests.Memote.Config.ConsistencyConfig","page":"Reference","title":"FBCModelTests.Memote.Config.ConsistencyConfig","text":"mutable struct ConsistencyConfig\n\nFields\n\nmass_ignored_reactions::Vector{String}\ncharge_ignored_reactions::Vector{String}\nconsistency_ignored_reactions::Vector{String}\n\n\n\n\n\n","category":"type"},{"location":"functions/#FBCModelTests.Memote.Config.EnergyConfig","page":"Reference","title":"FBCModelTests.Memote.Config.EnergyConfig","text":"mutable struct EnergyConfig\n\nFields\n\natpm_strings::Vector{String}\nenergy_dissipating_metabolites::Dict{String, String}\nadditional_energy_generating_reactions::Vector{COBREXA.Reaction}\nignored_energy_reactions::Vector{String}\noptimizer_modifications::Vector{Function}\n\n\n\n\n\n","category":"type"},{"location":"functions/#FBCModelTests.Memote.Config.GPRAssociationConfig","page":"Reference","title":"FBCModelTests.Memote.Config.GPRAssociationConfig","text":"mutable struct GPRAssociationConfig\n\nFields\n\ntest_annotation::String\n\n\n\n\n\n","category":"type"},{"location":"functions/#FBCModelTests.Memote.Config.MemoteConfig","page":"Reference","title":"FBCModelTests.Memote.Config.MemoteConfig","text":"mutable struct MemoteConfig\n\nA grouping of parameters used by the metabolic testing infrastructure.\n\nFields\n\nannotation::FBCModelTests.Memote.Config.AnnotationConfig\nbasic::FBCModelTests.Memote.Config.BasicConfig\nbiomass::FBCModelTests.Memote.Config.BiomassConfig\nconsistency::FBCModelTests.Memote.Config.ConsistencyConfig\nenergy::FBCModelTests.Memote.Config.EnergyConfig\ngpra::FBCModelTests.Memote.Config.GPRAssociationConfig\nmetabolite::FBCModelTests.Memote.Config.MetaboliteConfig\nnetwork::FBCModelTests.Memote.Config.NetworkConfig\nreaction::FBCModelTests.Memote.Config.ReactionConfig\n\n\n\n\n\n","category":"type"},{"location":"functions/#FBCModelTests.Memote.Config.MetaboliteConfig","page":"Reference","title":"FBCModelTests.Memote.Config.MetaboliteConfig","text":"mutable struct MetaboliteConfig\n\nFields\n\nformula_corner_cases::Vector{String}\ncharge_corner_cases::Vector{Int64}\nmedium_only_imported::Bool\ntest_annotation::String\n\n\n\n\n\n","category":"type"},{"location":"functions/#FBCModelTests.Memote.Config.NetworkConfig","page":"Reference","title":"FBCModelTests.Memote.Config.NetworkConfig","text":"mutable struct NetworkConfig\n\nFields\n\ncondition_number::Float64\nfva_bound::Float64\ncycle_tol::Float64\nminimum_metabolite_flux::Float64\noptimizer_modifications::Vector{Function}\n\n\n\n\n\n","category":"type"},{"location":"functions/#FBCModelTests.Memote.Config.ReactionConfig","page":"Reference","title":"FBCModelTests.Memote.Config.ReactionConfig","text":"mutable struct ReactionConfig\n\nFields\n\ntest_annotation::String\nignore_annotations::Vector{String}\nbound_default::Float64\n\n\n\n\n\n","category":"type"},{"location":"functions/#MEMOTE-checks","page":"Reference","title":"MEMOTE checks","text":"","category":"section"},{"location":"functions/#Basic-checks","page":"Reference","title":"Basic checks","text":"","category":"section"},{"location":"functions/","page":"Reference","title":"Reference","text":"Modules = [FBCModelTests.Memote.Basic]\nPages = [\"src/memote/checks/Basic.jl\"]","category":"page"},{"location":"functions/#FBCModelTests.Memote.Basic","page":"Reference","title":"FBCModelTests.Memote.Basic","text":"module Basic\n\nBasic metabolic model tests.\n\n\n\n\n\n","category":"module"},{"location":"functions/#FBCModelTests.Memote.Basic.model_compartments-Tuple{COBREXA.MetabolicModel}","page":"Reference","title":"FBCModelTests.Memote.Basic.model_compartments","text":"model_compartments(model::COBREXA.MetabolicModel) -> Set\n\n\nReturn the number of unique compartments in the model.\n\n\n\n\n\n","category":"method"},{"location":"functions/#FBCModelTests.Memote.Basic.model_has_compartments-Tuple{COBREXA.MetabolicModel}","page":"Reference","title":"FBCModelTests.Memote.Basic.model_has_compartments","text":"model_has_compartments(\n    model::COBREXA.MetabolicModel\n) -> Bool\n\n\nTest if the model has one or more compartments.\n\n\n\n\n\n","category":"method"},{"location":"functions/#FBCModelTests.Memote.Basic.model_has_genes-Tuple{COBREXA.MetabolicModel}","page":"Reference","title":"FBCModelTests.Memote.Basic.model_has_genes","text":"model_has_genes(model::COBREXA.MetabolicModel) -> Any\n\n\nTest if the model has any genes.\n\n\n\n\n\n","category":"method"},{"location":"functions/#FBCModelTests.Memote.Basic.model_has_metabolites-Tuple{COBREXA.MetabolicModel}","page":"Reference","title":"FBCModelTests.Memote.Basic.model_has_metabolites","text":"model_has_metabolites(model::COBREXA.MetabolicModel) -> Any\n\n\nTest if the model has any metabolites.\n\n\n\n\n\n","category":"method"},{"location":"functions/#FBCModelTests.Memote.Basic.model_has_name-Tuple{COBREXA.MetabolicModel}","page":"Reference","title":"FBCModelTests.Memote.Basic.model_has_name","text":"model_has_name(model::COBREXA.MetabolicModel) -> Bool\n\n\nTest if the model has an id. Note, this test will fail unless you use a StandardModel and its id field is assigned.\n\n\n\n\n\n","category":"method"},{"location":"functions/#FBCModelTests.Memote.Basic.model_has_reactions-Tuple{COBREXA.MetabolicModel}","page":"Reference","title":"FBCModelTests.Memote.Basic.model_has_reactions","text":"model_has_reactions(model::COBREXA.MetabolicModel) -> Any\n\n\nTest if the model has any reactions.\n\n\n\n\n\n","category":"method"},{"location":"functions/#FBCModelTests.Memote.Basic.model_metabolic_coverage-Tuple{COBREXA.MetabolicModel}","page":"Reference","title":"FBCModelTests.Memote.Basic.model_metabolic_coverage","text":"model_metabolic_coverage(\n    model::COBREXA.MetabolicModel\n) -> Any\n\n\nCalculate the metabolic coverage by dividing the number of reactions by the number of genes.\n\n\n\n\n\n","category":"method"},{"location":"functions/#FBCModelTests.Memote.Basic.model_metabolic_coverage_exceeds_minimum-Tuple{COBREXA.MetabolicModel}","page":"Reference","title":"FBCModelTests.Memote.Basic.model_metabolic_coverage_exceeds_minimum","text":"model_metabolic_coverage_exceeds_minimum(\n    model::COBREXA.MetabolicModel;\n    config\n) -> Any\n\n\nTest if the metabolic coverage, calculated with model_metabolic_coverage, exceeds config.basic.minimum_metabolic_coverage.\n\n\n\n\n\n","category":"method"},{"location":"functions/#FBCModelTests.Memote.Basic.model_solves_in_default_medium-Tuple{COBREXA.MetabolicModel, Any}","page":"Reference","title":"FBCModelTests.Memote.Basic.model_solves_in_default_medium","text":"model_solves_in_default_medium(\n    model::COBREXA.MetabolicModel,\n    optimizer;\n    config\n) -> Any\n\n\nCheck if the model can be solved under default conditions and yield a reasonable growth rate. Here reasonable is set via config.basic.minimum_growth_rate and config.basic.maximum_growth_rate. Optionally, pass optimization modifications to the solver through config.basic.optimizer_modifications.\n\n\n\n\n\n","category":"method"},{"location":"functions/#Consistency-checks","page":"Reference","title":"Consistency checks","text":"","category":"section"},{"location":"functions/","page":"Reference","title":"Reference","text":"Modules = [FBCModelTests.Memote.Consistency]\nPages = [\"src/memote/checks/Consistency.jl\"]","category":"page"},{"location":"functions/#FBCModelTests.Memote.Consistency","page":"Reference","title":"FBCModelTests.Memote.Consistency","text":"module Consistency\n\nThis module checks if the metabolic model is overall consistent.\n\n\n\n\n\n","category":"module"},{"location":"functions/#FBCModelTests.Memote.Consistency.model_is_consistent-Tuple{COBREXA.MetabolicModel, Any}","page":"Reference","title":"FBCModelTests.Memote.Consistency.model_is_consistent","text":"model_is_consistent(\n    model::COBREXA.MetabolicModel,\n    optimizer;\n    config\n) -> Bool\n\n\nDetermines if the model is stoichiometrically consistent. Note, stoichiometric consistency does not guarantee that mass balances must hold in the model. A more robust check is reactions_mass_unbalanced, but this works if not all metabolites have mass assigned to them. Based on Gevorgyan, Albert, Mark G. Poolman, and David A. Fell. \"Detection of stoichiometric inconsistencies in biomolecular models.\" Bioinformatics (2008). Optionally ignore some reactions in this analysis by adding reaction IDs to config.consistency.consistency_ignored_reactions.\n\n\n\n\n\n","category":"method"},{"location":"functions/#FBCModelTests.Memote.Consistency.reactions_charge_unbalanced-Tuple{COBREXA.MetabolicModel}","page":"Reference","title":"FBCModelTests.Memote.Consistency.reactions_charge_unbalanced","text":"reactions_charge_unbalanced(\n    model::COBREXA.MetabolicModel;\n    config\n) -> Vector{String}\n\n\nIterates through all the reactions in model and checks if the charges across each reaction balance. Returns a list of reaction IDs that are charge unbalanced. Optionally, use config.consistency.mass_ignored_reactions to pass a vector of reaction ids to ignore in this process. Internally biomass and exchang reactions are ignored.\n\n\n\n\n\n","category":"method"},{"location":"functions/#FBCModelTests.Memote.Consistency.reactions_mass_unbalanced-Tuple{COBREXA.MetabolicModel}","page":"Reference","title":"FBCModelTests.Memote.Consistency.reactions_mass_unbalanced","text":"reactions_mass_unbalanced(\n    model::COBREXA.MetabolicModel;\n    config\n) -> Vector{String}\n\n\nIterates through all the reactions in model and checks if the mass across each reaction balances. Returns a list of reaction IDs that are mass unbalanced. Optionally, use config.consistency.charge_ignored_reactions to pass a vector of reaction ids to ignore in this process. Internally biomass and exchang reactions are ignored.\n\n\n\n\n\n","category":"method"},{"location":"functions/#Metabolite-checks","page":"Reference","title":"Metabolite checks","text":"","category":"section"},{"location":"functions/","page":"Reference","title":"Reference","text":"Modules = [FBCModelTests.Memote.Metabolite]\nPages = [\"src/memote/checks/Metabolite.jl\"]","category":"page"},{"location":"functions/#FBCModelTests.Memote.Metabolite","page":"Reference","title":"FBCModelTests.Memote.Metabolite","text":"module Metabolite\n\nA module testing various metabolite properties.\n\n\n\n\n\n","category":"module"},{"location":"functions/#FBCModelTests.Memote.Metabolite.metabolites_are_duplicated-Tuple{COBREXA.MetabolicModel, Any, Any}","page":"Reference","title":"FBCModelTests.Memote.Metabolite.metabolites_are_duplicated","text":"metabolites_are_duplicated(\n    model::COBREXA.MetabolicModel,\n    m1,\n    m2;\n    config\n) -> Any\n\n\nTest if metabolites m1 and m2 are different by comparing their config.metabolite.test_annotation field in the annotations of each metabolite. Note, if no annotations are present for one or both of the metabolites, then return true.\n\n\n\n\n\n","category":"method"},{"location":"functions/#FBCModelTests.Memote.Metabolite.metabolites_duplicated_in_compartment-Tuple{COBREXA.MetabolicModel}","page":"Reference","title":"FBCModelTests.Memote.Metabolite.metabolites_duplicated_in_compartment","text":"metabolites_duplicated_in_compartment(\n    model::COBREXA.MetabolicModel;\n    config\n) -> Dict{String, Set{String}}\n\n\nReturn a dictionary of metabolites that are duplicated in their compartment.\n\n\n\n\n\n","category":"method"},{"location":"functions/#FBCModelTests.Memote.Metabolite.metabolites_medium_components-Tuple{COBREXA.MetabolicModel}","page":"Reference","title":"FBCModelTests.Memote.Metabolite.metabolites_medium_components","text":"metabolites_medium_components(\n    model::COBREXA.MetabolicModel;\n    config\n) -> Vector{String}\n\n\nReturn a list of all boundary reactions that allow flux into the model and create a metabolite. Assume that boundary reactions only create a single metabolite. Use the testing config, config.metabolite.only_imported = false, to also return metabolites that can be produced by the model under default conditions.\n\n\n\n\n\n","category":"method"},{"location":"functions/#FBCModelTests.Memote.Metabolite.metabolites_no_charge-Tuple{COBREXA.MetabolicModel}","page":"Reference","title":"FBCModelTests.Memote.Metabolite.metabolites_no_charge","text":"metabolites_no_charge(\n    model::COBREXA.MetabolicModel;\n    config\n) -> Any\n\n\nList all metabolites without a charge. Use config.metabolite.charge_corner_cases to specify an extra case to check for charge's that are not properly assigned.\n\n\n\n\n\n","category":"method"},{"location":"functions/#FBCModelTests.Memote.Metabolite.metabolites_no_formula-Tuple{COBREXA.MetabolicModel}","page":"Reference","title":"FBCModelTests.Memote.Metabolite.metabolites_no_formula","text":"metabolites_no_formula(\n    model::COBREXA.MetabolicModel;\n    config\n) -> Any\n\n\nList all metabolites without a formula. Use config.metabolite.formula_corner_cases to specify an extra case to check for formula's that are not properly assigned.\n\n\n\n\n\n","category":"method"},{"location":"functions/#FBCModelTests.Memote.Metabolite.metabolites_unique-Tuple{COBREXA.MetabolicModel}","page":"Reference","title":"FBCModelTests.Memote.Metabolite.metabolites_unique","text":"metabolites_unique(\n    model::COBREXA.MetabolicModel;\n    config\n) -> Set{String}\n\n\nReturn a list of unique metabolites in model. Uses metabolites_are_duplicated internally and forwards test_annotation to it. The latter argument is used to determine if two metabolites are the same by checking for any correspondence.\n\n\n\n\n\n","category":"method"},{"location":"functions/#Reaction-checks","page":"Reference","title":"Reaction checks","text":"","category":"section"},{"location":"functions/","page":"Reference","title":"Reference","text":"Modules = [FBCModelTests.Memote.Reaction]\nPages = [\"src/memote/checks/Reaction.jl\"]","category":"page"},{"location":"functions/#FBCModelTests.Memote.Reaction","page":"Reference","title":"FBCModelTests.Memote.Reaction","text":"module Reaction\n\nA module testing reaction properties.\n\n\n\n\n\n","category":"module"},{"location":"functions/#FBCModelTests.Memote.Reaction._is_constrained-Tuple{Any, Any, Any}","page":"Reference","title":"FBCModelTests.Memote.Reaction._is_constrained","text":"_is_constrained(lb, ub, default) -> Union{Missing, Bool}\n\n\nTest if a reaction is constrained.\n\n\n\n\n\n","category":"method"},{"location":"functions/#FBCModelTests.Memote.Reaction.duplicate_reactions-Tuple{COBREXA.MetabolicModel}","page":"Reference","title":"FBCModelTests.Memote.Reaction.duplicate_reactions","text":"duplicate_reactions(\n    model::COBREXA.MetabolicModel\n) -> Set{String}\n\n\nReturn a list of all reactions that are duplicated.\n\n\n\n\n\n","category":"method"},{"location":"functions/#FBCModelTests.Memote.Reaction.find_all_purely_metabolic_reactions-Tuple{COBREXA.MetabolicModel}","page":"Reference","title":"FBCModelTests.Memote.Reaction.find_all_purely_metabolic_reactions","text":"find_all_purely_metabolic_reactions(\n    model::COBREXA.MetabolicModel;\n    config\n) -> Tuple{Set{String}, Set{String}}\n\n\nFind all purely metabolic reactions and return two sets, one with unconstrained metabolic reactions (directional constraints do not count as constraints in this sense), and one with constrained metabolic reactions. For a constraint to be not purely directional, the lower or upper bound needs to be different from [-config.reactions.bound_default, 0, config.reactions.bound_default]. Metabolic reactions exclude transport, boundary and biomass reactions.\n\n\n\n\n\n","category":"method"},{"location":"functions/#FBCModelTests.Memote.Reaction.find_all_transport_reactions-Tuple{COBREXA.MetabolicModel}","page":"Reference","title":"FBCModelTests.Memote.Reaction.find_all_transport_reactions","text":"find_all_transport_reactions(\n    model::COBREXA.MetabolicModel;\n    config\n) -> Tuple{Set{String}, Set{String}}\n\n\nFind all transport reactions and and return two sets, one with unconstrained transport reactions (directional constraints do not count as constraints in this sense), and one with constrained transport reactions. For a constraint to be not purely directional, the lower or upper bound needs to be different from [-config.reactions.bound_default, 0, config.reactions.bound_default]. Transport reactions are heuristically identified, see Memote.Utils._probably_transport_reaction, which partially uses reaction annotations. Set the annotation field to use via config.reactions.rest_annotation.\n\n\n\n\n\n","category":"method"},{"location":"functions/#FBCModelTests.Memote.Reaction.reactions_with_identical_genes-Tuple{COBREXA.MetabolicModel}","page":"Reference","title":"FBCModelTests.Memote.Reaction.reactions_with_identical_genes","text":"reactions_with_identical_genes(\n    model::COBREXA.MetabolicModel\n) -> Dict{Vector{String}, Set{String}}\n\n\nIdentify reactions that have the same gene reaction rules. Does not take directionality or compartments into account.\n\n\n\n\n\n","category":"method"},{"location":"functions/#FBCModelTests.Memote.Reaction.reactions_with_partially_identical_annotations-Tuple{COBREXA.MetabolicModel}","page":"Reference","title":"FBCModelTests.Memote.Reaction.reactions_with_partially_identical_annotations","text":"reactions_with_partially_identical_annotations(\n    model::COBREXA.MetabolicModel;\n    config\n) -> Set{String}\n\n\nFind all reactions with overlapping annotation information. Internally calls COBREXA.annotation_index. Some annotations, like sbo terms will necessarily be non-unique, ignore annotations like this by editing config.reaction.ignore_annotations.\n\n\n\n\n\n","category":"method"},{"location":"functions/#GPRAssociation-checks","page":"Reference","title":"GPRAssociation checks","text":"","category":"section"},{"location":"functions/","page":"Reference","title":"Reference","text":"Modules = [FBCModelTests.Memote.GPRAssociation]\nPages = [\"src/memote/checks/GPRAssociation.jl\"]","category":"page"},{"location":"functions/#FBCModelTests.Memote.GPRAssociation","page":"Reference","title":"FBCModelTests.Memote.GPRAssociation","text":"module GPRAssociation\n\nA module testing various facets of the gene reaction associations.\n\n\n\n\n\n","category":"module"},{"location":"functions/#FBCModelTests.Memote.GPRAssociation.reactions_transport_no_gpr-Tuple{COBREXA.MetabolicModel}","page":"Reference","title":"FBCModelTests.Memote.GPRAssociation.reactions_transport_no_gpr","text":"reactions_transport_no_gpr(\n    model::COBREXA.MetabolicModel;\n    config\n) -> Any\n\n\nReturn a list of transport reactions that do not have gene reaction rules assigned to them.\n\n\n\n\n\n","category":"method"},{"location":"functions/#FBCModelTests.Memote.GPRAssociation.reactions_with_complexes-Tuple{COBREXA.MetabolicModel}","page":"Reference","title":"FBCModelTests.Memote.GPRAssociation.reactions_with_complexes","text":"reactions_with_complexes(\n    model::COBREXA.MetabolicModel\n) -> Any\n\n\nReturn a list of reaction ids that have protein complexes assigned to them.\n\n\n\n\n\n","category":"method"},{"location":"functions/#FBCModelTests.Memote.GPRAssociation.reactions_without_gpr-Tuple{COBREXA.MetabolicModel}","page":"Reference","title":"FBCModelTests.Memote.GPRAssociation.reactions_without_gpr","text":"reactions_without_gpr(model::COBREXA.MetabolicModel) -> Any\n\n\nReturn a list of reaction ids that do not have gene reaction rules (aka gene protein reaction associations).\n\n\n\n\n\n","category":"method"},{"location":"functions/#Biomass-checks","page":"Reference","title":"Biomass checks","text":"","category":"section"},{"location":"functions/","page":"Reference","title":"Reference","text":"Modules = [FBCModelTests.Memote.Biomass]\nPages = [\"src/memote/checks/Biomass.jl\"]","category":"page"},{"location":"functions/#FBCModelTests.Memote.Biomass","page":"Reference","title":"FBCModelTests.Memote.Biomass","text":"module Biomass\n\nThis module contains tests that check the consistency of the biomass reaction.\n\n\n\n\n\n","category":"module"},{"location":"functions/#FBCModelTests.Memote.Biomass.atp_present_in_biomass-Tuple{COBREXA.MetabolicModel}","page":"Reference","title":"FBCModelTests.Memote.Biomass.atp_present_in_biomass","text":"atp_present_in_biomass(\n    model::COBREXA.MetabolicModel;\n    config\n) -> Any\n\n\nCheck if the biomass reaction consumes ATP and H₂O, and produces ADP, HO₄P, and H⁺. Each of these metabolites have a lookup table mapping them to the name space of the model, defined in config.biomass.growth_metabolites. These need to be set if you use anything other than the BiGG namespace.\n\n\n\n\n\n","category":"method"},{"location":"functions/#FBCModelTests.Memote.Biomass.biomass_missing_essential_precursors-Tuple{COBREXA.MetabolicModel}","page":"Reference","title":"FBCModelTests.Memote.Biomass.biomass_missing_essential_precursors","text":"biomass_missing_essential_precursors(\n    model::COBREXA.MetabolicModel;\n    config\n) -> Dict{String, Vector{String}}\n\n\nTests if each biomass reaction contains a set of essential precursors, listed in config.biomass.essential_precursors. Note, this function only works on a lumped biomass function.\n\n\n\n\n\n","category":"method"},{"location":"functions/#FBCModelTests.Memote.Biomass.find_blocked_biomass_precursors-Tuple{COBREXA.MetabolicModel, Any}","page":"Reference","title":"FBCModelTests.Memote.Biomass.find_blocked_biomass_precursors","text":"find_blocked_biomass_precursors(\n    model::COBREXA.MetabolicModel,\n    optimizer;\n    config\n) -> Dict{String, Vector{String}}\n\n\nCheck if the model can synthesize all of the biomass precursors in all of the biomass functions, except those listed in config.biomass.ignored_precursors in the default medium. Set any optimizer modifications with config.biomass.optimizer_modifications.\n\n\n\n\n\n","category":"method"},{"location":"functions/#FBCModelTests.Memote.Biomass.model_biomass_is_consistent-Tuple{COBREXA.MetabolicModel}","page":"Reference","title":"FBCModelTests.Memote.Biomass.model_biomass_is_consistent","text":"model_biomass_is_consistent(\n    model::COBREXA.MetabolicModel;\n    config\n) -> Any\n\n\nCheck that the molar mass of each biomass reactions falls within [1 - 1e-3, 1 + 1e-6] by calling model_biomass_molar_mass internally.\n\n\n\n\n\n","category":"method"},{"location":"functions/#FBCModelTests.Memote.Biomass.model_biomass_molar_mass-Tuple{COBREXA.MetabolicModel}","page":"Reference","title":"FBCModelTests.Memote.Biomass.model_biomass_molar_mass","text":"model_biomass_molar_mass(\n    model::COBREXA.MetabolicModel;\n    config\n) -> Any\n\n\nFor each biomass reaction, identified by model_biomass_reactions, calculate the molar weight of the reaction by summing the products of the associated metabolite coefficients with their molar masses.\n\n\n\n\n\n","category":"method"},{"location":"functions/#FBCModelTests.Memote.Biomass.model_biomass_reactions-Tuple{COBREXA.MetabolicModel}","page":"Reference","title":"FBCModelTests.Memote.Biomass.model_biomass_reactions","text":"model_biomass_reactions(\n    model::COBREXA.MetabolicModel;\n    config\n) -> Set\n\n\nIdentify all the biomass reactions in the model using both sbo annotations, as well biomass strings typically contained in their reaction IDs. Use config.biomass.biomass_strings to update the list of strings to look for.\n\n\n\n\n\n","category":"method"},{"location":"functions/#Network-checks","page":"Reference","title":"Network checks","text":"","category":"section"},{"location":"functions/","page":"Reference","title":"Reference","text":"Modules = [FBCModelTests.Memote.Network]\nPages = [\"src/memote/checks/Network.jl\"]","category":"page"},{"location":"functions/#FBCModelTests.Memote.Network","page":"Reference","title":"FBCModelTests.Memote.Network","text":"module Network\n\nA module testing the network and topology properties of the model.\n\n\n\n\n\n","category":"module"},{"location":"functions/#FBCModelTests.Memote.Network._find_orphan_or_deadend_metabolites-Tuple{COBREXA.MetabolicModel}","page":"Reference","title":"FBCModelTests.Memote.Network._find_orphan_or_deadend_metabolites","text":"_find_orphan_or_deadend_metabolites(\n    model::COBREXA.MetabolicModel;\n    consumed\n) -> Vector{String}\n\n\nHelper function to find orphan or deadend metabolites. Specify consumed=true to consider orphan metabolites or false to consider deadend metabolites. Set complete_medium=true to open all boundary reactions to simulate a complete medium.\n\n\n\n\n\n","category":"method"},{"location":"functions/#FBCModelTests.Memote.Network.find_all_universally_blocked_reactions-Tuple{COBREXA.MetabolicModel, Any}","page":"Reference","title":"FBCModelTests.Memote.Network.find_all_universally_blocked_reactions","text":"find_all_universally_blocked_reactions(\n    model::COBREXA.MetabolicModel,\n    optimizer;\n    config\n) -> Any\n\n\nMake all boundary reactions reversible and run FVA on the model to find all reactions that are universally blocked. Optimizer modifications can be passed through config.network.optimizer_modifications\n\n\n\n\n\n","category":"method"},{"location":"functions/#FBCModelTests.Memote.Network.find_complete_medium_orphans_and_deadends-Tuple{COBREXA.MetabolicModel, Any}","page":"Reference","title":"FBCModelTests.Memote.Network.find_complete_medium_orphans_and_deadends","text":"find_complete_medium_orphans_and_deadends(\n    model::COBREXA.MetabolicModel,\n    optimizer;\n    config\n) -> Dict{Symbol, Vector{String}}\n\n\nReturn a dictionary mapping orphan and deadend metabolites that occur in the model in complete medium. Complete medium is modeled by opening all the bounday reactions. For each metabolite FBA is run with a temporary reaction either consuming or producing the metabolite in question. At minimum a flux of config.network.minimum_metabolite_flux must be attained for the metabolite to pass the test.\n\n\n\n\n\n","category":"method"},{"location":"functions/#FBCModelTests.Memote.Network.find_cycle_reactions-Tuple{COBREXA.MetabolicModel, Any}","page":"Reference","title":"FBCModelTests.Memote.Network.find_cycle_reactions","text":"find_cycle_reactions(\n    model::COBREXA.MetabolicModel,\n    optimizer;\n    config\n) -> Set{String}\n\n\nFind all reactions that participate in stoichiometrically balanced cycles by closing all boundary reactions and running fva on the resultant model.\n\n\n\n\n\n","category":"method"},{"location":"functions/#FBCModelTests.Memote.Network.find_deadend_metabolites-Tuple{COBREXA.MetabolicModel}","page":"Reference","title":"FBCModelTests.Memote.Network.find_deadend_metabolites","text":"find_deadend_metabolites(\n    model::COBREXA.MetabolicModel\n) -> Vector{String}\n\n\nFind all metabolites that can only (excludes reversible reactions) be produced in the model by inspecting the stoichiometric matrix.\n\n\n\n\n\n","category":"method"},{"location":"functions/#FBCModelTests.Memote.Network.find_orphan_metabolites-Tuple{COBREXA.MetabolicModel}","page":"Reference","title":"FBCModelTests.Memote.Network.find_orphan_metabolites","text":"find_orphan_metabolites(\n    model::COBREXA.MetabolicModel\n) -> Vector{String}\n\n\nFind all metabolites that can only (excludes reversible reactions) be consumed in the model by inspecting the stoichiometric matrix.\n\n\n\n\n\n","category":"method"},{"location":"functions/#FBCModelTests.Memote.Network.stoichiometric_matrix_is_well_conditioned-Tuple{COBREXA.MetabolicModel}","page":"Reference","title":"FBCModelTests.Memote.Network.stoichiometric_matrix_is_well_conditioned","text":"stoichiometric_matrix_is_well_conditioned(\n    model::COBREXA.MetabolicModel;\n    config\n) -> Any\n\n\nTest if the stoichiometric matrix is well conditioned by determining if stoichiometric_max_min_ratio is less than 10⁹ (which can be set in config.network.condition_number).\n\n\n\n\n\n","category":"method"},{"location":"functions/#FBCModelTests.Memote.Network.stoichiometric_max_min_ratio-Tuple{COBREXA.MetabolicModel}","page":"Reference","title":"FBCModelTests.Memote.Network.stoichiometric_max_min_ratio","text":"stoichiometric_max_min_ratio(\n    model::COBREXA.MetabolicModel\n) -> Any\n\n\nReturn the ratio of the absolute maximum and minimum value of the nonzero coefficients in the stoichiometric matrix of model.\n\n\n\n\n\n","category":"method"},{"location":"functions/#Annotation-checks","page":"Reference","title":"Annotation checks","text":"","category":"section"},{"location":"functions/","page":"Reference","title":"Reference","text":"Modules = [FBCModelTests.Memote.Annotation]\nPages = [\"src/memote/checks/Annotation.jl\"]","category":"page"},{"location":"functions/#FBCModelTests.Memote.Annotation","page":"Reference","title":"FBCModelTests.Memote.Annotation","text":"module Annotation\n\nThis module contains tests used to check the coverage and conformance of reaction, gene, and metabolite annotations.\n\n\n\n\n\n","category":"module"},{"location":"functions/#FBCModelTests.Memote.Annotation._find_nonconformal_components-Tuple{COBREXA.MetabolicModel, Any, Any, Any}","page":"Reference","title":"FBCModelTests.Memote.Annotation._find_nonconformal_components","text":"_find_nonconformal_components(\n    model::COBREXA.MetabolicModel,\n    id_accessor,\n    annotation_accessor,\n    annotation_regex\n) -> Dict{String, Vector{String}}\n\n\nHelper function to find all the annotations that do not conform in the model.\n\n\n\n\n\n","category":"method"},{"location":"functions/#FBCModelTests.Memote.Annotation._find_unannotated_components-Tuple{COBREXA.MetabolicModel, Any, Any, Any}","page":"Reference","title":"FBCModelTests.Memote.Annotation._find_unannotated_components","text":"_find_unannotated_components(\n    model::COBREXA.MetabolicModel,\n    id_accessor,\n    annotation_accessor,\n    annotation_kws\n) -> Dict{String, Vector{String}}\n\n\nHelper function to find all the unannotated components in the model.\n\n\n\n\n\n","category":"method"},{"location":"functions/#FBCModelTests.Memote.Annotation.find_all_unannotated_genes-Tuple{COBREXA.MetabolicModel}","page":"Reference","title":"FBCModelTests.Memote.Annotation.find_all_unannotated_genes","text":"find_all_unannotated_genes(\n    model::COBREXA.MetabolicModel\n) -> Any\n\n\nChecks if every gene has an annotation and returns an array of genes that do not have annotatons.\n\n\n\n\n\n","category":"method"},{"location":"functions/#FBCModelTests.Memote.Annotation.find_all_unannotated_metabolites-Tuple{COBREXA.MetabolicModel}","page":"Reference","title":"FBCModelTests.Memote.Annotation.find_all_unannotated_metabolites","text":"find_all_unannotated_metabolites(\n    model::COBREXA.MetabolicModel\n) -> Any\n\n\nChecks if every metabolite has an annotation and returns an array of metabolites that do not have annotatons.\n\n\n\n\n\n","category":"method"},{"location":"functions/#FBCModelTests.Memote.Annotation.find_all_unannotated_reactions-Tuple{COBREXA.MetabolicModel}","page":"Reference","title":"FBCModelTests.Memote.Annotation.find_all_unannotated_reactions","text":"find_all_unannotated_reactions(\n    model::COBREXA.MetabolicModel\n) -> Any\n\n\nChecks if every reaction has an annotation and returns an array of reactions that do not have annotatons.\n\n\n\n\n\n","category":"method"},{"location":"functions/#FBCModelTests.Memote.Annotation.find_database_unannotated_genes-Tuple{Any}","page":"Reference","title":"FBCModelTests.Memote.Annotation.find_database_unannotated_genes","text":"find_database_unannotated_genes(\n    model;\n    config\n) -> Dict{String, Vector{String}}\n\n\nChecks if the databases listed in config.annotation.gene_annotation_keywords are present in the gene annotations. Returns a dictionary of annotation keywords mapped to a list of genes that do not include the keyword.\n\n\n\n\n\n","category":"method"},{"location":"functions/#FBCModelTests.Memote.Annotation.find_database_unannotated_metabolites-Tuple{Any}","page":"Reference","title":"FBCModelTests.Memote.Annotation.find_database_unannotated_metabolites","text":"find_database_unannotated_metabolites(\n    model;\n    config\n) -> Dict{String, Vector{String}}\n\n\nChecks if the databases listed in config.annotation.metabolite_annotation_keywords are present in the metabolite annotations. Returns a dictionary of annotation keywords mapped to a list of metabolites that do not include the keyword.\n\n\n\n\n\n","category":"method"},{"location":"functions/#FBCModelTests.Memote.Annotation.find_database_unannotated_reactions-Tuple{Any}","page":"Reference","title":"FBCModelTests.Memote.Annotation.find_database_unannotated_reactions","text":"find_database_unannotated_reactions(\n    model;\n    config\n) -> Dict{String, Vector{String}}\n\n\nChecks if the databases listed in config.annotation.reaction_annotation_keywords are present in the reaction annotations. Returns a dictionary of annotation keywords mapped to a list of reactions that do not include the keyword.\n\n\n\n\n\n","category":"method"},{"location":"functions/#FBCModelTests.Memote.Annotation.find_nonconformal_gene_annotations-Tuple{Any}","page":"Reference","title":"FBCModelTests.Memote.Annotation.find_nonconformal_gene_annotations","text":"find_nonconformal_gene_annotations(\n    model;\n    config\n) -> Dict{String, Vector{String}}\n\n\nCheck if the gene annotation entry conforms to commonly recognized formats of annotation database using regex patterns. Uses the database formats listed in config.annotation.gene_annotation_regexes to test the conformity. Returns a dictionary mapping the database id to a list of genes that do not conform.\n\n\n\n\n\n","category":"method"},{"location":"functions/#FBCModelTests.Memote.Annotation.find_nonconformal_metabolite_annotations-Tuple{Any}","page":"Reference","title":"FBCModelTests.Memote.Annotation.find_nonconformal_metabolite_annotations","text":"find_nonconformal_metabolite_annotations(\n    model;\n    config\n) -> Dict{String, Vector{String}}\n\n\nCheck if the metabolite annotation entry conforms to commonly recognized formats of annotation database using regex patterns. Uses the database formats listed in config.annotation.metabolite_annotation_regexes to test the conformity. Returns a dictionary mapping the database id to a list of metabolites that do not conform.\n\n\n\n\n\n","category":"method"},{"location":"functions/#FBCModelTests.Memote.Annotation.find_nonconformal_reaction_annotations-Tuple{Any}","page":"Reference","title":"FBCModelTests.Memote.Annotation.find_nonconformal_reaction_annotations","text":"find_nonconformal_reaction_annotations(\n    model;\n    config\n) -> Dict{String, Vector{String}}\n\n\nCheck if the reaction annotation entry conforms to commonly recognized formats of annotation database using regex patterns. Uses the database formats listed in config.annotation.reaction_annotation_regexes to test the conformity. Returns a dictionary mapping the database id to a list of reactions that do not conform.\n\n\n\n\n\n","category":"method"},{"location":"functions/#Energy-checks","page":"Reference","title":"Energy checks","text":"","category":"section"},{"location":"functions/","page":"Reference","title":"Reference","text":"Modules = [FBCModelTests.Memote.Energy]\nPages = [\"src/memote/checks/Energy.jl\"]","category":"page"},{"location":"functions/#FBCModelTests.Memote.Energy","page":"Reference","title":"FBCModelTests.Memote.Energy","text":"module Energy\n\nThis module checks if the model is energetically sensible.\n\n\n\n\n\n","category":"module"},{"location":"functions/#FBCModelTests.Memote.Energy.model_has_atpm_reaction-Tuple{COBREXA.MetabolicModel}","page":"Reference","title":"FBCModelTests.Memote.Energy.model_has_atpm_reaction","text":"model_has_atpm_reaction(\n    model::COBREXA.MetabolicModel;\n    config\n) -> Union{Missing, Bool}\n\n\nCheck if model has an ATP maintenance reaction built in (also called a non-growth associated maintenance cost). Looks for reaction annotations corresponding to the sbo maintenance term, or looks for reaction ids that contain the strings listed in config.energy.atpm_strings.\n\n\n\n\n\n","category":"method"},{"location":"functions/#FBCModelTests.Memote.Energy.model_has_no_erroneous_energy_generating_cycles-Tuple{COBREXA.MetabolicModel, Any}","page":"Reference","title":"FBCModelTests.Memote.Energy.model_has_no_erroneous_energy_generating_cycles","text":"model_has_no_erroneous_energy_generating_cycles(\n    model::COBREXA.MetabolicModel,\n    optimizer;\n    config\n) -> Bool\n\n\nAttempts to detect if the model contains any erroneous energy generating cycles by closing all exchange reactions and using flux balance analysis to maximize the sum of fluxes through a set of energy dissipating reactions. The flux sum should be zero if the model is free of energy generating reactions. This function is based on Fritzemeier, Claus Jonathan, et al. \"Erroneous energy-generating cycles in published genome scale metabolic networks: Identification and removal.\" PLoS computational biology (2017). The energy dissipating reactions are based on the source paper, and include:\n\nATP + H2O --> ADP + H + Phosphate\nCTP + H2O --> CDP + H + Phosphate\nGTP + H2O --> GDP + H + Phosphate\nUTP + H2O --> UDP + H + Phosphate\nITP + H2O --> IDP + H + Phosphate\nNADH --> H + NAD\nNADPH --> H + NADP\nFADH2 --> 2 H + FAD\nFMNH2 --> 2 H + FMN\nUbiquinol-8 --> 2 H + Ubiquinone-8\nMenaquinol-8 --> 2 H + Menaquinone-8\n2-Demethylmenaquinol-8 --> 2 H + 2-Demethylmenaquinone-8\nH2O + ACCOA --> H + Acetate + COA\nL-Glutamate + H2O --> 2-Oxoglutarate + Ammonium + 2 H\nH[external] --> H\n\nAdditional energy dissipating reactions can be directly specified through config.energy.additional_energy_generating_reactions, which should be vector of COBREXA Reactions using the same metabolite name space as the model. Internally, the model is converted to a COBREXA StandardModel, so ensure that the appropriate accessors are defined for it. Since models use different name spaces, config.energy.energy_dissipating_metabolites is used to create the energy dissipating reactions. By default it uses the BiGG name space, but this will be changed to ChEBI in due course. If your model uses a different name space, then you have to change the values (NOT the keys) of config.energy.energy_dissipating_metabolites. Each energy dissipating reaction is added to the test only if all its associated metabolites are present. Any config.energy.optimizer_modifications to the solver are passed directly through to COBREXA's flux_balance_analysis function. All config.energy.boundary_reactions and config.energy.ignored_energy_reactions are deleted from an internal copy of model; this internal copy is used for analysis. Returns true if the model has no energy generating cycles.\n\n\n\n\n\n","category":"method"},{"location":"functions/#Types-and-utilities","page":"Reference","title":"Types and utilities","text":"","category":"section"},{"location":"functions/","page":"Reference","title":"Reference","text":"Modules = [FBCModelTests]\nPages = [\"common.jl\"]","category":"page"},{"location":"#FBCModelTests.jl-—-testing-and-reproducibility-of-constraint-based-metabolic-modeling","page":"Home","title":"FBCModelTests.jl — testing and reproducibility of constraint-based metabolic modeling","text":"","category":"section"},{"location":"#Table-of-contents","page":"Home","title":"Table of contents","text":"","category":"section"},{"location":"","page":"Home","title":"Home","text":"Pages = [\"functions.md\"]\nDepth = 2","category":"page"}]
}
